### PR TITLE
ci: remove environment from release and update publishing doc

### PR DESCRIPTION
**Pull request by randi274**
### What does this PR do?
The environment during the `Test, Build, and Release` GHA was proving unnecessary. We can't get the VSIX files from the held build, so our updated flow should be: 
- `Test, Build, and Release` creates a github release without intervention
-  `Publish in Microsoft Marketplace` job starts automatically and pauses execution to allow for testing
-  VSIX files are downloaded from there for testing
-  If a serious bug is found, we can delete the tag & release
-  Otherwise, approve the `Publish in Microsoft Marketplace` job

### What issues does this PR fix or reference?
Follow-up to #4627 

### Functionality Before
- `Test, Build, and Release` **needs to be manually approved before creating a release**
-  `Publish in Microsoft Marketplace` job **starts after approval** and pauses execution to allow for testing
-  VSIX files are downloaded from there for testing
-  If a serious bug is found, we can delete the tag & release
-  Otherwise, approve the `Publish in Microsoft Marketplace` job

### Functionality After
- `Test, Build, and Release` creates a github release without intervention
-  `Publish in Microsoft Marketplace` job starts automatically and pauses execution to allow for testing
-  VSIX files are downloaded from there for testing
-  If a serious bug is found, we can delete the tag & release
-  Otherwise, approve the `Publish in Microsoft Marketplace` job

[skip-pr-validation]

### CODE REVIEW

File: contributing/publishing.md [hunk 1]
```diff
@@ -45,27 +45,28 @@ The PreRelease job will verify if the version of the branch to be merged is newe
...
-Before approving the release to the marketplace, download the vsix files, install them locally and verify they are working as expected.
+Before approving the release to the marketplace, download the vsix files from the release you just created, install them locally and verify they are working as expected.
+
+Alternatively, you can download the files using the [gh cli](https://cli.github.com/) and then upload them all at once. Replace `v57.3.0` with the tag name for the release that you are testing, and to whatever download directory you would like. Additionally, `code` can be replaced by `code-insiders`.
+
+`> gh release download v57.3.0 --dir ~/Downloads/v57.3.0 --pattern '*.vscode'`  
+`> find ~/Downloads/v53.3.0 -type f -name "*.vsix" -exec code --install-extension {} \;`
 
-After you feel comfortable publishing to the marketplace, approve the publish job in the GitHub Action UI.
+After completing your release testing following our internal template, approve the publish job on your .
```
Comment (CristiCanizales) [inline contributing/publishing.md:63]: is the final word maybe missing here? 😝

Comment (randi274) [reply contributing/publishing.md:63]: good catch! let me review and add that 😄

File: contributing/publishing.md [hunk 2]
```diff
@@ -45,27 +45,28 @@ The PreRelease job will verify if the version of the branch to be merged is newe
...
-Before approving the release to the marketplace, download the vsix files, install them locally and verify they are working as expected.
+Before approving the release to the marketplace, download the vsix files from the release you just created, install them locally and verify they are working as expected.
+
+Alternatively, you can download the files using the [gh cli](https://cli.github.com/) and then upload them all at once. Replace `v57.3.0` with the tag name for the release that you are testing, and to whatever download directory you would like. Additionally, `code` can be replaced by `code-insiders`.
+
+`> gh release download v57.3.0 --dir ~/Downloads/v57.3.0 --pattern '*.vscode'`  
+`> find ~/Downloads/v53.3.0 -type f -name "*.vsix" -exec code --install-extension {} \;`
 
-After you feel comfortable publishing to the marketplace, approve the publish job in the GitHub Action UI.
+After completing your release testing following our internal template, approve the publish job "Publish in Microsoft Marketplace" to allow the extensions to be uploaded to the marketplace and complete the release process.
```
Comment (randi274) [inline contributing/publishing.md:63]: @CristiCanizales - completed my unfinished sentence 🎉

**Review by gbockus-sf — APPROVED** [review]
🚢

**Comment by gbockus-sf** [comment]
FYI You'll need a ticket to associate this PR with in order to get past the PR validator.

**Pull request merged**